Implementation/core/User.py

```diff
@@ -0,0 +1,53 @@
+from data.Database import connect_to_db
+from service.Validation import Validation
+from utils.logger import log_activity
+from core.Exception import DatabaseConnectionError, InvalidInputError
+
+class User:
+    def __init__(self, user_id, user_name, user_role):
+        self.user_id = user_id
+        self.user_name = user_name
+        self.user_role = user_role
+
+    @staticmethod
+    def login(user_id, user_password):
+        try:
+            conn = connect_to_db()
```

Comment:
Multiple things are happening at different levels here. Try and wrap into separate functions.